### PR TITLE
Implement MessageDatabase and Firebase corresponding object

### PR DESCRIPTION
**Pull request by MarcelTorne**
This is the first step towards resolving issue #46

### CODE REVIEW

File: app/src/main/java/com/github/epfl/meili/messages/FirebaseMessageDatabaseAdapter.kt
Comment (Bwananah) [inline app/src/main/java/com/github/epfl/meili/messages/FirebaseMessageDatabaseAdapter.kt:44]: Is it necessary to override without replacing it with anything ?

File: app/src/androidTest/java/com/github/epfl/meili/messages/FirebaseMessageDatabaseAdapterTest.kt
Comment (Bwananah) [inline app/src/androidTest/java/com/github/epfl/meili/messages/FirebaseMessageDatabaseAdapterTest.kt:22]: What is MOCK_PATH ? Is it used to create a mock database ?

**Review by Bwananah — COMMENTED** [review]
Very good job. You are a machine ! I would prefer if you added more comments to explain your code as I am not as familiar with android or Firebase. But I should get better at reading code as the semester continues.

I am waiting until you see my questions before I approve, but it should be fine

File: app/src/main/java/com/github/epfl/meili/messages/FirebaseMessageDatabaseAdapter.kt
Comment (MarcelTorne) [reply app/src/main/java/com/github/epfl/meili/messages/FirebaseMessageDatabaseAdapter.kt:44]: Unfortunately yes it is

File: app/src/androidTest/java/com/github/epfl/meili/messages/FirebaseMessageDatabaseAdapterTest.kt
Comment (MarcelTorne) [reply app/src/androidTest/java/com/github/epfl/meili/messages/FirebaseMessageDatabaseAdapterTest.kt:22]: I will write it as comment, path is the path of the Firebase database, so yes it is for a mock database

**Review by Bwananah — APPROVED** [review]
Thanks for the changes, good job !

**Review by ahmedc3ca — APPROVED** [review]
A great addition to the code. I will need to rebase this to my branch to use the firebaseMessageAdapter.

**Pull request merged**
